File: 2020/day1/day1_part1.go
package main

import (
    "fmt"
    "io/ioutil"
    "strconv"
    "strings"
)

func main() {
    content, err := ioutil.ReadFile("input.txt")

    if err != nil {
      panic(err)
    }

    lines := strings.Split(string(content), "\n")

    set := make(map[int]bool)
    set[0] = true

    for _, l := range lines {
      number, _ := strconv.Atoi(l)
      number_to_check := 2020 - number

      if set[number_to_check] {
        fmt.Printf("%v\n", number_to_check * number)
        break
      }

      set[number] = true
    }
}

File: 2020/day1/day1_part1.rb
input = File.open('input.txt')

lines = input.read.split

checked_nums = { 0 => true }

lines.each do |li|
  num = li.to_i
  check_num = 2020 - num

  if checked_nums[check_num]
    puts "#{check_num} + #{num} = 2020"
    puts "#{check_num} * #{num} = #{check_num * num}"

    break
  end

  checked_nums[num] = true
end
